puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/builders/foreman-plugins-pull-request.yaml

```
@@ -54,12 +54,16 @@
           # Update dependencies
           bundle update --jobs=5 --retry=5
 
-          # we need to install node modules for integration tests
-          npm install
-
           # Now let's add the plugin migrations
           bundle exec rake db:migrate
 
+          # If the plugin contains integration tests or triggers core integration tests,
+          # we need to install node modules and compile webpack
+          if [ -d ${{PLUGIN_ROOT}}/test/integration ] || [ ${{database}} = postgresql ]; then
+            npm install
+            bundle exec rake webpack:compile
```

Perhaps we should change this:
```suggestion
            tasks="jenkins:integration ${{tasks}}"
```

Of course also move `tasks="jenkins:unit"` to before the if condition. Then we can drop the `[ ${{database}} = postgresql ]` part since that obviously has no effect.

---

dropping the db condition would cause running core integration tests as well on plugins for mysql and sqlite.

---

Hmm, that's also ugly.

---

How about using `tasks="webpack:compile ${{tasks}}"` instead? That would prevent duplicate compilation on `jenkins:integration` (which your PR does). Then for postgres it becomes `bundle exec rake webpack:compile jenkins:unit jenkins:integration`.

---

good idea, updated accordingly and also changed it for the plugin tests